packages/ui-utils/src/is-defined/index.js

```diff
@@ -0,0 +1,13 @@
+/* eslint-disable jsdoc/valid-types */
+/**
+ * Checks whether a value is defined, asserting that it is neither
+ * null nor undefined.
+ *
+ * @template T
+ * @param {T | undefined | null} value The value to check.
+ * @return {value is T} Whether the value is undefined.
+ */
+const isDefined = ( value ) => value !== null && value !== undefined;
```

Comment:
This already exists in lodash `isNil`

Reply:
You're right, but if we'd like to remove `lodash` at scale, as described by https://github.com/WordPress/gutenberg/issues/17025, we'd need to replace this utility (it's a quite useful one, if you search through G2 for `is.defined` you'll see it's used almost everywhere).

We could replace it with `! isNil` but that doesn't accomplish getting rid of lodash at all.

Reply:
I'm not sure I agree with "getting rid of lodash by adding  our own utils". I think getting rid of lodash by using native functions is definitely a good thing but other than that:

 - Lodash can be  replaced with lodash-es for npm consumers
 - Lodash is only loaded once in WordPress, it's better that bundling the same utils multiple times.

Reply:
Sounds good to me. We'll replace `is.defined` with `! isNil` everywhere then?